### PR TITLE
MPI flexible Hello World test

### LINKED ISSUE #179: Add a flexible MPI hello test

### PR DESCRIPTION
**Pull request by rsarm**
Closes #179

### CODE REVIEW

File: cscs-checks/microbenchmarks/mpi/mpi_helloworld.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        self.maintainers = ['RS', 'VK']
+        self.num_tasks_per_node = 1
+        self.num_tasks = 0
+        num_processes = sn.extractsingle(
+            r'Received messages from (?P<nprocs>\d+) processes',
+            self.stdout, 'nprocs', int)
+        self.sanity_patterns = sn.assert_eq(num_processes, self.real_num_tasks)
+
+    @property
+    @deferrable
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/mpi/mpi_helloworld.py]: The `@deferrable` decorator is for internal use. You should `@sn.sanity_function` instead.

File: cscs-checks/microbenchmarks/mpi/mpi_helloworld.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    def real_num_tasks(self):
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/mpi/mpi_helloworld.py]: Better call this `num_tasks_assigned`.

File: cscs-checks/microbenchmarks/mpi/mpi_helloworld.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class MPIHelloWorldTest(RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__('mpi_helloworld_check',
+                         os.path.dirname(__file__), **kwargs)
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc',
+                              'kech:cn', 'kech:pn',
+                              'leone:normal', 'monch:compute']
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/mpi/mpi_helloworld.py]: Have you tested it in all those systems? CI won't do that for you, cos it's not a production test.

File: cscs-checks/microbenchmarks/mpi/mpi_helloworld.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        self.descr = 'MPI Hello World'
+        self.sourcepath = 'mpi_helloworld.c'
+        self.executable = './mpi_helloworld'
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/mpi/mpi_helloworld.py]: This is not needed.

File: cscs-checks/microbenchmarks/mpi/mpi_helloworld.py (hunk 3 above)
Comment (rsarm) [reply cscs-checks/microbenchmarks/mpi/mpi_helloworld.py]: I didn't realize that. `kesch` was misspelled.

**Comment by rsarm** [comment]
@jenkins-cscs retry all

**Pull request merged**